### ISSUE TITLE
Fastlane  deliver download_metadata alway says 'The login credentials for '***@***.com' seem to be wrong Do you want to re-enter your password? (y/n)' forever 

### ISSUE THREAD

**Issue by EricHYF** [fastlane/fastlane#27457]
# Case

When I run this in terminal

```
fastlane deliver download_metadata -u ${USER_NAME} -a ${APP_ID}
```
it alway says:
```
The login credentials for '***@***.com' seem to be wrong Do you want to re-enter your password? (y/n)
```
And after I retry several times, My apple ID has been locked

I've checked that My Apple ID and password is correct. 

Realy need help to resolve the problerm, THX

See the details:

---

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                      |
| --------------------------- | ---------------------------------------------------------- |
| OS                          | 14.2.1                                                     |
| Ruby                        | 3.3.5                                                      |
| Bundler?                    | false                                                      |
| Git                         | git version 2.39.3 (Apple Git-146)                         |
| Installation Source         | /opt/homebrew/Cellar/fastlane/2.225.0/libexec/bin/fastlane |
| Host                        | macOS 14.2.1 (23C71)                                       |
| Ruby Lib Dir                | /opt/homebrew/Cellar/ruby/3.3.5/lib                        |
| OpenSSL Version             | OpenSSL 3.3.1 4 Jun 2024                                   |
| Is contained                | false                                                      |
| Is homebrew                 | true                                                       |
| Is installed via Fabric.app | false                                                      |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                |
| Xcode Version               | 15.4                                                       |
| Swift Version               | 5.10                                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | zh_CN.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby

default_platform(:ios)

platform :ios do

    # the download cmd
    desc "download app metadata"
    lane :download_metadata do
        # shellString = "fastlane deliver download_metadata -a #{ENV["APP_ID"]} -u #{ENV["USER_NAME"]}" 
 	    
        
      
        api_key = app_store_connect_api_key(
            key_id: "5AX32B569D",
            issuer_id: "586c6fff-a8dd-4747-8d6f-d1f29b0b8925",
            key_filepath: "./Jenkins_CI_GuruSmart_AuthKey_5AX32B569D.p8"
        )
        shellString = "fastlane deliver download_metadata -a #{ENV["APP_ID"]} -u #{ENV["USER_NAME"]}"
        sh("#{shellString}")
        
    end
   
    desc "upload app metadata"
    lane :upload_metadata do
        deliver(
            username: "#{ENV["USER_NAME"]}",
            app: "#{ENV["APP"]}",
            app_identifier: "#{ENV["APP_ID"]}",
            team_id: "#{ENV["TEAM_ID"]}",
            skip_binary_upload: true,
            skip_screenshots:true,
            skip_metadata:false,
            skip_app_version_update: true,
            force: true,
        )
    end
end
```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.225.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| error_highlight                 | 0.6.0        |
| did_you_mean                    | 1.6.3        |
| syntax_suggest                  | 2.0.0        |
| public_suffix                   | 6.0.1        |
| addressable                     | 2.8.7        |
| artifactory                     | 3.0.17       |
| jmespath                        | 1.6.2        |
| aws-partitions                  | 1.992.0      |
| aws-eventstream                 | 1.3.0        |
| aws-sigv4                       | 1.10.0       |
| aws-sdk-core                    | 3.210.0      |
| aws-sdk-kms                     | 1.95.0       |
| aws-sdk-s3                      | 1.169.0      |
| babosa                          | 1.0.4        |
| bundler                         | 2.5.18       |
| rexml                           | 3.3.8        |
| nkf                             | 0.1.3        |
| base64                          | 0.2.0        |
| CFPropertyList                  | 3.0.7        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| dotenv                          | 2.8.1        |
| emoji_regex                     | 3.2.3        |
| excon                           | 0.112.0      |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| multipart-post                  | 2.4.1        |
| faraday-multipart               | 1.0.4        |
| faraday-net_http                | 1.0.2        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| faraday-retry                   | 1.0.3        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.10.4       |
| faraday_middleware              | 1.2.1        |
| domain_name                     | 0.6.20240107 |
| http-cookie                     | 1.0.7        |
| faraday-cookie_jar              | 0.0.7        |
| fastimage                       | 2.3.1        |
| sysrandom                       | 1.0.5        |
| fastlane-sirp                   | 1.0.0        |
| gh_inspector                    | 1.1.3        |
| uber                            | 0.1.0        |
| declarative                     | 0.0.20       |
| trailblazer-option              | 0.1.2        |
| representable                   | 3.2.0        |
| retriable                       | 3.1.2        |
| mini_mime                       | 1.1.5        |
| jwt                             | 2.9.3        |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| signet                          | 0.19.0       |
| googleauth                      | 1.8.1        |
| httpclient                      | 2.8.3        |
| google-apis-core                | 0.11.3       |
| google-apis-androidpublisher_v3 | 0.54.0       |
| google-apis-playcustomapp_v1    | 0.13.0       |
| google-cloud-env                | 1.6.0        |
| google-cloud-errors             | 1.4.0        |
| google-cloud-core               | 1.7.1        |
| google-apis-iamcredentials_v1   | 0.17.0       |
| google-apis-storage_v1          | 0.31.0       |
| rake                            | 13.2.1       |
| digest-crc                      | 0.6.5        |
| google-cloud-storage            | 1.47.0       |
| json                            | 2.7.1        |
| mini_magick                     | 4.13.2       |
| naturally                       | 2.2.1        |
| optparse                        | 0.4.0        |
| plist                           | 3.7.1        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.5        |
| simctl                          | 1.6.10       |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 2.6.0        |
| terminal-table                  | 3.0.2        |
| tty-screen                      | 0.8.2        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| atomos                          | 0.1.3        |
| claide                          | 1.1.0        |
| colored2                        | 3.1.2        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.25.1       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
| set                             | 1.1.0        |
| forwardable                     | 1.3.3        |
| logger                          | 1.6.0        |
| pathname                        | 0.3.0        |
| shellwords                      | 0.2.0        |
| cgi                             | 0.4.1        |
| date                            | 3.3.4        |
| timeout                         | 0.4.1        |
| stringio                        | 3.1.1        |
| securerandom                    | 0.3.1        |
| uri                             | 0.13.1       |
| ipaddr                          | 1.2.6        |
| openssl                         | 3.2.0        |
| digest                          | 3.1.1        |
| io-nonblock                     | 0.3.0        |
| zlib                            | 3.1.1        |
| resolv                          | 0.3.0        |
| io-wait                         | 0.3.1        |
| time                            | 0.3.0        |
| open-uri                        | 0.4.1        |
| mutex_m                         | 0.2.0        |
| net-http                        | 0.4.1        |
| net-protocol                    | 0.2.2        |
| ostruct                         | 0.6.0        |
| english                         | 0.8.0        |
| erb                             | 4.0.3        |
| strscan                         | 3.0.9        |
| abbrev                          | 0.1.2        |
| io-console                      | 0.7.1        |
| tempfile                        | 0.2.1        |
| delegate                        | 0.3.1        |
| fileutils                       | 1.7.2        |
| tmpdir                          | 0.2.0        |
| singleton                       | 0.2.0        |
| open3                           | 0.2.1        |
| prettyprint                     | 0.2.0        |
| pp                              | 0.5.0        |
| find                            | 0.2.0        |
| yaml                            | 0.3.0        |
| psych                           | 5.1.2        |
</details>

*generated on:* **2024-11-04**
</details>

**Comment by EricHYF** [comment]
# PS
I've try to write the command in fastfile. but it still get the same issue. SAD...

**Comment by TalRandi1** [comment]
Same behavior for me when I tried running - 

fastlane spaceauth -u <username>

My account has been locked recently and it seems to be related. Anyway, It's unlocked now and I managed to log into my account via the browser.

**Comment by alanflickgames** [comment]
I have had the same error since last week when trying to use fastlane deliver. My account is _not_ locked out as I can still upload builds via Transporter and add metadata via the browser.  
Only Fastlane fails to login, and it fails regardless of whether I use my regular password or an app-specific password. 

I've tried specifying the password in both a DeliverFile in my fastlane project, and manually specifying them when deliver is called:
```
export FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD=abcd-abcd-abcd-abcd
export FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT=1 
export FASTLANE_ITUNES_TRANSPORTER_PATH=/usr/local/itms
export FASTLANE_PASSWORD=abcd-abcd-abcd-abcd

fastlane deliver
```

**Comment by EricHYF** [comment]
> Same behavior for me when I tried running -
> 
> fastlane spaceauth -u
> 
> My account has been locked recently and it seems to be related. Anyway, It's unlocked now and I managed to log into my account via the browser.

Saddly, my account is still been locked after I'm trying to run the command and failedseveral times....
BTW, I can login via browser and unlock my account.
But isn't it ridiculous to have to use a browser to unlock your account every time?

**Comment by alanflickgames** [comment]
Update: I was able to get Fastlane to successfully login.  
First I had to reset my Apple password, I'm not sure why though as it worked elsewhere. The next login attempt (with new password set as an environment variable) still failed, but it then asked me to re-enter my password which then worked.
  
So something is not right somewhere, but it's possible that it's an issue at Apple's end. In my case the old password was one that had not been changed in a _very_ long time, so it's possible that was the cause of it being rejected when used in a CLI (even though it did not state that as the reason for failure).

**Comment by TalRandi1** [comment]
@alanflickgames, 
Thanks for the update.
Unfortunately, I tried changing the password, and it didn't work. 

If anyone has any more suggestions, I'd love to try.

**Comment by Evgeniyberkut** [comment]
So i experienced the same issue
After a full day of researching and considering to do as [MattKiazyk](https://github.com/fastlane/fastlane/pull/27461) offered i found the solution for me:

- I simply changed apple id from "Example@icloud.com" to "example@icloud.com" in my project and it worked.

**Comment by EricHYF** [comment]
An update on my recent progress:
The issue has been solved.

The developer account I use for myself hasn't updated its password in over 3 years
This seems to be considered a password vulnerability by Apple.
After trying various options to no avail
I changed the password for my developer AppleID.
Fastlane now works fine.

I hope this provides some ideas for those of you who are experiencing the same problem as me.
Thanks to those who replied to my post!

The problem can be closed now!
